coderd/promoauth/oauth2.go

```diff
@@ -36,11 +37,23 @@ var _ OAuth2Config = (*Config)(nil)
 // Primarily to avoid any prometheus errors registering duplicate metrics.
 type Factory struct {
 	metrics *metrics
+	// optional replace now func
+	Now func() time.Time
 }
 
 // metrics is the reusable metrics for all oauth2 providers.
 type metrics struct {
 	externalRequestCount *prometheus.CounterVec
+
+	// if the oauth supports it, rate limit metrics.
+	// rateLimit is the defined limit per interval
+	rateLimit          *prometheus.GaugeVec
+	rateLimitRemaining *prometheus.GaugeVec
+	rateLimitUsed      *prometheus.GaugeVec
+	// rateLimitReset is the time in seconds the rate limit resets.
```

Comment:
Is this the unix time? I ask because this and the description below are very similar. Could perhaps be clarified.

Reply:
Yea, I'll clarify